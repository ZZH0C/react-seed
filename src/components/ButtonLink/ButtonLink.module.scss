.button_home {
  box-sizing: border-box;
  position: absolute;
  top: 0;
  right: 0;

  border-radius: 2px;
  border: 1px solid transparent;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.24), 0 0 1px 0 rgba(0, 0, 0, 0.24);
  color: rgba(0, 0, 0, 0.54);
  margin: 1rem ;
  height: 4rem;
  width: 10rem;
  background: white;
  text-decoration: none;
  padding: 0;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.button_home:hover {
  text-decoration: none;
  box-shadow: 0 0.2rem 0.2rem 0 rgba(0, 0, 0, 0.5);
}

.button_home:active {
  text-decoration: none;
  box-shadow: inset 0 0.2rem 0.2rem 0 rgba(0, 0, 0, 1);
}

.button_home:focus{
  text-decoration: none;
}