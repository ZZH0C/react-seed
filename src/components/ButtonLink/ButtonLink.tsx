import React from 'react';
import { Link } from 'react-router-dom';
import styles from './ButtonLink.module.scss';
import classNames from 'classnames';

interface ButtonLinkProps extends React.ButtonHTMLAttributes<HTMLElement> {
  href: string;
}

export const ButtonLink: React.FC<ButtonLinkProps> = ({ href, children }) => {
  return (
    <Link className={classNames(styles.button_home)} to={href}>
      {children}
    </Link>
  );
};
