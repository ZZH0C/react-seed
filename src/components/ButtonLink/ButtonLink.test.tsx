import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { ButtonLink } from './ButtonLink';

describe('components/ButtonLink', () => {
  const mockProps = {
    href: '/',
    children: 'some text',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot', () => {
    expect(
      shallow(
        <ButtonLink href={mockProps.href}>{mockProps.children}</ButtonLink>,
      ),
    ).toMatchSnapshot();
  });
});
