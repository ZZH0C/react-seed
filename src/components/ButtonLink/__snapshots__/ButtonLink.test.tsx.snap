// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ButtonLink should render and match snapshot 1`] = `
<Link
  className="button_home"
  to="/"
>
  some text
</Link>
`;
